### ISSUE TITLE
Issue with gigangic screen (very probably *not* an astroid bug per se but Xembed?)

### ISSUE THREAD

**Issue by hugoroy** [astroidmail/astroid#175]
I don't know if it's a bug in how Astroid uses Xembed or if it's a bug in Xembed per se, but this is how emacs integration looks in astroid when using:
- a gigantic screen (we're talking 4K)
- setting a scaling factor of 2 with `gsettings set org.gnome.desktop.interface scaling-factor 2`

I don't know where it is most adequate to report this

![capture d ecran de 2016-07-13 19-27-58](https://cloud.githubusercontent.com/assets/1629771/16813082/5ca5bf8c-4930-11e6-8d60-2d07e527dc5c.png)
.

**Comment by gauteh** [comment]
Does it happen with gvim? What happens if you try to resize the window
after the editor is started? Do you have anything about text-width being
80 chars anywhere in your config or in the format of emacs?

**Comment by hugoroy** [comment]
Yes, it does happen with gvim. Although with gvim, resizing the window after it starts will solve the issue (not with emacs).

Yes I probably have stuff in .emacs affecting this, for instance:

```
(when window-system (set-frame-size (selected-frame) 80 40))
```

**Comment by gauteh** [comment]
Hugo Roy writes on juli 14, 2016 18:38:
> Yes, it does happen with gvim. Although with gvim, resizing the window after it starts will solve the issue (not with emacs).

Is there some remote command that can be sent to make emacs update? Then
this could be run when the window is resized. It would be better if we
could avoid that though.

> Yes I probably have stuff in .emacs affecting this, for instance:
> 
> ```
> (when window-system (set-frame-size (selected-frame) 80 40))
> ```

Maybe you can do something about that, you could maybe send some custom
commands to the emacs command that you launch from astroid if you need
to.

**Comment by hugoroy** [comment]
It was only a screen I grabbed at a friends' office. But I'll try to test when I get my hands on another gigantic screen.

**Comment by hugoroy** [comment]
I now have a gigantic screen myself so I can test :-)

Starting emacs with a blank ".emacs" triggered the same issue. So after all it does not seem that there's anything special on my side of emacs that would cause this.

I don't see anything special in the logs either.

**Comment by hugoroy** [comment]
I guess the problem really is with Xembed. Here's some weird stuff going on: you can see the window underneath showing:

![capture d ecran de 2016-07-27 00-16-17](https://cloud.githubusercontent.com/assets/1629771/17157424/69276742-538f-11e6-9247-9c1652f09cdc.png)

**Comment by gauteh** [comment]
This might be an GTK bug too - I would think it is the place to report the bug anyway, does the plug-socket example from GTK produce the same results? https://developer.gnome.org/gtkmm-tutorial/stable/sec-plugs-sockets-example.html.en or https://github.com/gauteh/plug-socket-grab-focus (instructions: https://bugzilla.gnome.org/show_bug.cgi?id=729248)

**Comment by hugoroy** [comment]
I am not sure I understand the connection. I did not run into any apparent probme (except that it was focused on the left, not on the right).
![capture d ecran de 2016-08-07 00-19-04](https://cloud.githubusercontent.com/assets/1629771/17459371/b0e082d2-5c34-11e6-86b5-37076451c463.png)

**Comment by gauteh** [comment]
Sorry, I thought that example would be useful - but it only works for
that specific example.

The idea was to have a simple socket program that you could add emacs to
and check if it was still a problem.

**Comment by gauteh** [comment]
What happens if you don't have a scaling factor?

**Comment by hugoroy** [comment]
Here is how it shows: https://storage.5apps.com/hugo/public/shares/160826-2225-Capture%20d%27%C3%A9cran%20vid%C3%A9o%20de%2027-08-2016%2000%3A24%3A10.webm

**Comment by gauteh** [comment]
Hugo Roy writes on august 27, 2016 0:26:
> Here is how it shows: https://storage.5apps.com/hugo/public/shares/160826-2225-Capture%20d%27%C3%A9cran%20vid%C3%A9o%20de%2027-08-2016%2000%3A24%3A10.webm 
> 

Ah, that is very useful! I suspect this is could be a bug in how gnome
propagates the window/screen/whatever size to X11. I think we should
submit this as a bug to GTK. I'll see if I can manage to reproduce it
here..

**Comment by hugoroy** [comment]
If I add the "-mm" (maximized) option, then the emacs frame will be maximized but only after resizing the whole astroid window. So yes, this may be only an emacs bug?

**Comment by gauteh** [comment]
Hugo Roy writes on september 11, 2016 23:09:
> If I add the "-mm" (maximized) option, then the emacs frame will be maximized but only after resizing the whole astroid window. So yes, this may be only an emacs bug?
> 

What does maximized mean? fullscreen? Should it make any difference when
the size of the emacs window is fixed to the embedding window?

**Comment by gauteh** [comment]
Merged #242, which can be used as an workaround for this. Closing.
